online-shop/src/Sagas/NewProductSaga/ManageNewProductSagaCalls.tsx

```diff
@@ -0,0 +1,44 @@
+import { put } from "redux-saga/effects";
+import { IProduct } from "../../Models/Models";
+import { createNewProduct, NPResetLoadingStatus } from "../../ReduxStore/NewProductSection/actions";
+
+export function* performNewProductRequest(action: any) {
+  let productId: number = 123;
+  let serverResponse: number = 0;
+  let newProduct: IProduct = action.payload;
+  const apiEndpoint = "http://localhost:4000/products";
+
+  //Activate loading window
+  yield put(NPResetLoadingStatus());
+
+  //Get the number of products in store, so we can generate an id for the new product
+  yield fetch(apiEndpoint, { method: 'GET' })
+    .then(response => response.json())
+    .then(result => {
+      productId = [...result].length + 1;
+    });
+
+  newProduct.id = productId;
+
+  //Call the server with the new product data, to create the product
+  yield fetch(apiEndpoint, {
```

Comment:
I would recommend extracting the fetch calls (+ corresponding then / catch) into service / API objects in a separate folder and just call these service methods here in the sagas. That way the saga looks cleaner.